packages/wasm/crate/src/tx.rs

```diff
@@ -278,13 +279,25 @@ pub async fn transaction_info_inner(
             Action::Swap(swap) => {
                 let commitment = swap.body.payload.commitment;
 
-                let swap_position_option = storage
-                    .get_swap_by_commitment(commitment.into())
-                    .await?
+                let swap_record = storage.get_swap_by_commitment(commitment.into()).await?;
+
+                let swap_position_option = swap_record
+                    .clone()
                     .map(|swap_record| Position::from(swap_record.position));
 
+                swap_record
+                    .clone()
+                    .and_then(|swap_record| swap_record.output_data)
+                    .map(|output_data| {
+                        if let Ok(bsod) = BatchSwapOutputData::try_from(output_data) {
+                            txp.batch_swap_output_data.push(bsod);
+                        }
+
+                        Some(())
+                    });
```

Comment:
Think this should be moved up above `let swap_position_option =...`. Also, can we get a code comment on what this is doing?

Reply:
What do you think about this?

```rust
       Action::Swap(swap) => {
                let commitment = swap.body.payload.commitment;
                if let Some(swap_record) = storage.get_swap_by_commitment(commitment.into()).await? {
                    // Add swap output to perspective
                    if let Some(output_data) = swap_record.output_data {
                        let bsod = BatchSwapOutputData::try_from(output_data)?;
                        txp.batch_swap_output_data.push(bsod)
                    }

                    // Add swap claim to perspective
                    let swap_position = Position::from(swap_record.position);
                    add_swap_claim_txn_to_perspective(
                        &storage,
                        &fvk,
                        &mut txp,
                        &commitment,
                        swap_position,
                    ).await?;
                }
            }

```

Reply:
Ah, much cleaner. Will update.